include/gunrock/cuda/launch_box.hxx

```diff
@@ -26,45 +29,29 @@ namespace launch_box {
 
 /**
  * @brief CUDA dim3 template representation, since dim3 cannot be used as a
- * template argument
- * @tparam x_ Dimension in the X direction
- * @tparam y_ Dimension in the Y direction
- * @tparam z_ Dimension in the Z direction
+ * template argument.
+ *
+ * @tparam x_ Dimension in the X direction.
+ * @tparam y_ (default = `1`) Dimension in the Y direction.
+ * @tparam z_ (default = `1`) Dimension in the Z direction.
  */
 template <unsigned int x_, unsigned int y_ = 1, unsigned int z_ = 1>
 struct dim3_t {
   enum : unsigned int { x = x_, y = y_, z = z_, size = x_ * y_ * z_ };
   static constexpr dim3 get_dim3() { return dim3(x, y, z); }
 };
 
-enum sm_flag_t : unsigned {
-  fallback = 0,
-  sm_30 = 1 << 0,
-  sm_35 = 1 << 1,
-  sm_37 = 1 << 2,
-  sm_50 = 1 << 3,
-  sm_52 = 1 << 4,
-  sm_53 = 1 << 5,
-  sm_60 = 1 << 6,
-  sm_61 = 1 << 7,
-  sm_62 = 1 << 8,
-  sm_70 = 1 << 9,
-  sm_72 = 1 << 10,
-  sm_75 = 1 << 11,
-  sm_80 = 1 << 12,
-  sm_86 = 1 << 13
-};
-
-// Macro for the flag of the current device's SM version
-#define SM_TARGET_FLAG _SM_FLAG_WRAPPER(SM_TARGET)
-// "ver" will be expanded before the call to _SM_FLAG
-#define _SM_FLAG_WRAPPER(ver) _SM_FLAG(ver)
-#define _SM_FLAG(ver) sm_##ver
+/**
+ * @brief Bit flag enum representing different SM architectures.
+ *
+ */
+enum sm_flag_t : unsigned;
 
 /**
- * @brief Overloaded bitwise OR operator
- * @param lhs Left-hand side
- * @param rhs Right-hand side
+ * @brief Overloaded bitwise OR operator.
+ *
+ * @param lhs Left-hand side.
+ * @param rhs Right-hand side.
  * \return sm_flag_t
  */
 constexpr sm_flag_t operator|(sm_flag_t lhs, sm_flag_t rhs) {
```

Comment:
If you end up creating that new file, you can move these sm-specific operators there as well.

Reply:
Done